### PR TITLE
OAK-9771: IndexWriter config should be customisable from oak

### PR DESCRIPTION
**Pull request by tihom88**

### CODE REVIEW

**Review by thomasmueller — COMMENTED** [review]
I think we don't need to document these settings as of now. Not documenting them has the advantage that people don't use them, and hence don't rely on them.

File: oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexConfigTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        System.setProperty("oak.index.lucene.maxBufferedDeleteTerms", "1000");
+        System.setProperty("oak.index.lucene.ramPerThreadHardLimitMB", "100");
+    }
+
+    @After
+    public void after() {
+        System.setProperties(properties);
+        for (File d : dirs) {
+            FileUtils.deleteQuietly(d);
+        }
```
Comment (fabriziofortino) [inline oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexConfigTest.java]: `dirs` does not get ever updated. Can we remove it or some code is missing?

File: oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexConfigTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    @Test
+    public void testIndexWriterConfig() throws Exception {
```
Comment (fabriziofortino) [inline oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexConfigTest.java]: not needed
```suggestion
    public void testIndexWriterConfig() {
```

File: oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexConfigTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        System.setProperty("oak.index.lucene.ramPerThreadHardLimitMB", "100");
+    }
+
+    @After
+    public void after() {
+        System.setProperties(properties);
+        for (File d : dirs) {
+            FileUtils.deleteQuietly(d);
+        }
+    }
```
Comment (fabriziofortino) [inline oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexConfigTest.java]: Could we use this Junit4 extension https://stefanbirkner.github.io/system-rules/#ProvideSystemProperty to safely deal with system properties?

**Review by fabriziofortino — APPROVED** [review]
LGTM. Left a few comments on the test class.

File: oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexConfigTest.java (hunk 3 above)
Comment (tihom88) [reply oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexConfigTest.java]: Done

**Pull request merged**